doc/community/developer-guide.md

```diff
@@ -135,3 +135,37 @@ with pytest.raises(ZeroDivisionError) as excinfo:
 ```{code-cell} ipython3
 assert str(excinfo.value) == "division by zero"
 ```
+
+## General SQL Clause for Multiple Database Dialects
+
+### Context
+
+As our codebase is expanding, we have noticed that we need to write SQL queries for different database dialects such as MySQL, PostgreSQL, SQLite, and more. Writing and maintaining separate queries for each database can be time-consuming and error-prone.
+
+To address this issue, we have created a general SQL clause that can be used across multiple database dialects. This clause will allow us to write a single SQL query that can be translated to different database dialects, then use it for calculating the metadata (e.g. metadata used by boxplot)
+
+In this document, we will provide an overview of the general SQL clause, explain how it works, and provide examples of how it can be used in our codebase. We will also include instructions on how to add support for additional database dialects.
+
+### Example
+
+We can use [SQLGlot](https://sqlglot.com/sqlglot.html) to build the general sql expressions
+
+```python
+from sqlglot import select, condition
+
+where = condition("x=1").and_("y=1")
+general_sql = select("*").from_("y").where(where).sql()
+```
+```python
+'SELECT x FROM y, z'
+```
+
+Then transpile to the sql which is supported by current connected dialect.
+
+Our `sql.connection.Connection._transpile_query` will automatically detect the dialect and transpile the SQL clause
+
+```python
+query = sql.connection.Connection._transpile_query(general_sql)
```

Comment:
does the transpile_query function behave differently if it receives a string vs a sqlglot object? I'd assume that passing a sqlglot object is more reliable since it should be easier to export to certain dialect as we're removing the initial parsing stage

Reply:
I did the experiment and figure out our current `transpile_query()` can only help us to solve the "`" (backtick) transformation issue

In order to see the different transpiled sql results under different dialects, we need to provide the original SQL Clause with original dialect, Something like this:

```
from sql import connection
from sqlalchemy import create_engine

# Prepare connection
conn = connection.Connection(engine=create_engine(url="sqlite://"))

# Prepare SQL clause based on duckdb syntax
input_sql = sqlglot.parse_one("SELECT TO_TIMESTAMP(1618088028295)", read="duckdb").sql()
print(f"{conn._get_curr_sqlglot_dialect()} transpiled result:")
conn._transpile_query(input_sql)

# sqlite transpiled result:
# 'SELECT UNIX_TO_TIME(1618088028295)'
```